tests/integration/test_integration.py

```diff
@@ -44,6 +54,34 @@ async def run_get_ca_certificate_action(ops_test: OpsTest, timeout: int = 60) ->
     return await ops_test.model.get_action_output(action_uuid=action.entity_id, wait=timeout)
 
 
+async def validate_vault_status(
+    expected_vault_status_code: int | List[int], ops_test: OpsTest, vault_client: hvac.Client
+):
+    assert ops_test.model
+    async with ops_test.fast_forward():
+        await ops_test.model.wait_for_idle(
+            apps=[APP_NAME],
+            status="blocked",
+            timeout=1000,
+        )
+    response = vault_client.sys.read_health_status()
+    if isinstance(expected_vault_status_code, list):
+        assert response.status_code in expected_vault_status_code
+    else:
+        assert response.status_code == expected_vault_status_code
+
+
+async def get_leader(app: Application) -> Unit:
+    leader = None
+    for unit in app.units:
+        assert isinstance(unit, Unit)
+        if await unit.is_leader_from_status():
+            leader = unit
```

Comment:
Why not return here? this would remove the need for the `leader` variable (nit)

Reply:
Good point. Likely just a leftover from extracting this part to a method.

Reply:
Fixed in https://github.com/canonical/vault-operator/pull/62/commits/064f5826382a90b8130fc244de46d60ca5e94e25